apps/design/backend/src/app.ts

```diff
@@ -139,43 +144,63 @@ function buildApi({ store }: { store: Store }) {
       electionId: Id;
     }): Promise<{ zipContents: Buffer; electionHash: string }> {
       const { election, layoutOptions } = store.getElection(input.electionId);
-      const { ballots, electionDefinition } = layOutAllBallots({
-        election,
-        isTestMode: true,
-        layoutOptions,
-      }).unsafeUnwrap();
 
       const zip = new JsZip();
 
-      for (const { precinctId, document, gridLayout } of ballots) {
-        const { ballotStyleId } = gridLayout;
-        const precinct = assertDefined(
-          getPrecinctById({ election, precinctId })
-        );
-        const pdf = renderDocumentToPdf(document);
-        const fileName = `ballot-${precinct.name.replaceAll(
-          ' ',
-          '_'
-        )}-${ballotStyleId}.pdf`;
-        zip.file(fileName, pdf);
-        pdf.end();
+      const ballotTypes: Array<[BallotType, string]> = [
+        [BallotType.Standard, 'precinct'],
+        [BallotType.Absentee, 'absentee'],
+      ];
+      const ballotModes: BallotMode[] = ['official', 'test', 'sample'];
```

Comment:
One technique that I've seen used to guarantee that an array like this is exhaustive is as follows:

```ts
const BALLOT_MODES = ['official', 'test', 'sample'] as const;
type BallotMode = typeof BALLOT_MODES[number];
```

Basically define the exhaustive array first and then define the type from that

Reply:
Good suggestion, thanks! d1054c3fd